### PR TITLE
1902 create task form

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+export class AlgorithmTaskRegionVisualizer extends EntitiesVisualizer {
+	_iconSrc: Style = new Style({
+		image: new Icon({
+			scale: 1,
+			src: 'assets/pinpoint-indicator.svg'
+		}),
+		zIndex: 100
+	});
+
+	selfIntersectMessage = 'Invalid Polygon (Self-Intersect)';
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: self intersect?

File: src/app/@ansyn/menu-items/public_api.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { SetAlgorithmTaskDrawIndicator } from './algorithms/actions/algorithms.actions';
```
Comment (asafMasa) [inline src/app/@ansyn/menu-items/public_api.ts]: combine "algorithms.actions"

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	region$ = this.store$.select(selectAlgorithmTaskRegion);
+
+	isActiveMap$ = this.store$.select(selectActiveMapId).pipe(
+		map((activeMapId: string): boolean => activeMapId === this.mapId),
+		distinctUntilChanged()
+	);
+
+	drawIndicator$ = this.store$.select(selectAlgorithmTaskDrawIndicator);
+
+	// @AutoSubscription
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: remove

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	onDispose() {
+		this.removeDrawInteraction();
+		super.onDispose();
+	}
+
+	featureStyle(feature: Feature, resolution) {
+		return this._iconSrc;
+	}
+
+	drawRegionOnMap(region: CaseRegionState): Observable<boolean> {
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: change CaseRegionState

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+		const coordinates = getPointByGeometry(<GeoJSON.GeometryObject>region).coordinates;
+		const id = 'algorithmTaskRegion';
+		const featureJson = turf.point(coordinates);
+		const entities = [{ id, featureJson }];
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: draw rectangle

File: confd/templates/production.tmpl [hunk 6]
```diff
@@ -1057,5 +1058,12 @@
       "callbackURL": "{{ getv "/ansyn/auth0/callbackURL" "http://localhost:4200/#/callback/" }}",
       "scope": "{{ getv "/ansyn/auth0/scope" "openid" }}"
   },
-  "ORIENTATIONS": ["Align North", "User Perspective", "Imagery Perspective"]
+  "ORIENTATIONS": ["Align North", "User Perspective", "Imagery Perspective"],
+  "algorithmsConfig": {
+	"alg_1" : {
```
[the remark this answers is not in the document]
Comment (ramtob) [reply confd/templates/production.tmpl]: done

File: confd/templates/production.tmpl [hunk 7]
```diff
@@ -1057,5 +1058,12 @@
...
+		"maxOverlays": 20,
+		"timeEstimationPerOverlayInMinutes": 10,
+		"sensorNames": [ "Landsat8L1G", "Sentinel2L1C" ]
+	}
```
[the remark this answers is not in the document]
Comment (ramtob) [reply confd/templates/production.tmpl]: done

File: src/app/@ansyn/menu-items/algorithms/components/algorithms/algorithms.component.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+		*ngSwitchCase="'table'"
+		(goto)="gotoPage($event)"
+	></ansyn-tasks-table-page>
+
+	<ansyn-tasks-form-page
+		*ngSwitchCase="'form'"
+		(goto)="gotoPage($event)"
+	></ansyn-tasks-form-page>
+
+	<p *ngSwitchDefault>
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/algorithms/algorithms.component.html]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.html [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+				[(ngModel)]="whichOverlays"
+				[value]="'displayed_overlays'"
+				name="whichOverlays3"
+				disabled
+			>Displayed overlays
+			</ansyn-radio>
+		</div>
+	</div>
+
+	<div>Note: this algorithm is only available for 20 overlays</div>
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.html]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+	constructor(
+		protected configService: AlgorithmsConfigService,
+		protected algorithmsService: AlgorithmsService,
+		public translate: TranslateService,
+		protected store$: Store<any>
+	) {
+	}
+
+	ngOnInit() {
+		console.log(this.algorithms);
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts]: done

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: done

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts (hunk 3 above)
Comment (ramtob) [reply src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: done

File: src/app/@ansyn/menu-items/public_api.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/menu-items/public_api.ts]: done (the "import" statement seems to be redundant)

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts (hunk 4 above)
Comment (ramtob) [reply src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+		return this.algorithms[this.algName];
+	}
+
+	get timeEstimation() {
+		return this.currentAlgorithm.timeEstimationPerOverlayInMinutes * this.task.overlays.length;
+	}
+
+	@AutoSubscription
+	getOverlays$: Observable<any[]> = this.store$.select(selectFavoriteOverlays).pipe(
+		tap((favoriteOverlays) => {
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts]: done

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts (hunk 5 above)
Comment (ramtob) [reply src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: done

File: confd/templates/production.tmpl [hunk 12]
```diff
@@ -374,7 +374,8 @@
     "translation": {
     	"default": {
 		 	"Sentinel-2A": "Sentinel 2A",
-		 	"alg_1": "secret name"
+		 	"alg_1": "secret name 1",
```
Comment (asafMasa) [inline confd/templates/production.tmpl]: name 1

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.spec.ts [hunk 13]
```diff
@@ -84,4 +50,51 @@ describe('TasksFormComponent', () => {
...
+				name: '21',
+				overlays: overlays,
+				masterOverlay: overlays[0],
+				region: {
+					type: 'Point'
+				}
+			}
+		});
+		it('should set empty message by default', () => {
+			console.log('hehe');
```
Comment (asafMasa) [inline src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.spec.ts]: remove console

File: src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts [hunk 14]
```diff
@@ -20,22 +19,31 @@ import { combineLatest } from 'rxjs/index';
 import { OpenLayersMap } from '../../open-layers-map/openlayers-map/openlayers-map';
 import Icon from 'ol/style/icon';
 import Style from 'ol/style/style';
+import Stroke from 'ol/style/stroke';
 import Feature from 'ol/feature';
+import { selectAlgorithmTaskRegionLength } from '../../../../menu-items/algorithms/reducers/algorithms.reducer';
```
Comment (asafMasa) [inline src/app/@ansyn/plugins/openlayers/visualizers/algorithms/algorithm-task-region.visualizer.ts]: libs (public api)

File: confd/templates/production.tmpl (hunk 12 above)
Comment (ramtob) [reply confd/templates/production.tmpl]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.spec.ts (hunk 13 above)
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.spec.ts]: done

**Pull request merged**